rust/src/dataset.rs

```diff
@@ -290,17 +300,22 @@ impl Dataset {
         };
 
         let mut manifest = Manifest::new(&schema, Arc::new(fragments));
-        manifest.version = latest_manifest.map_or(1, |m| m.version + 1);
-        if matches!(params.mode, WriteMode::Overwrite) {
-            // If overwrite, invalidate index
-            manifest.index_section = None;
-        }
+        manifest.version = dataset.as_ref().map_or(1, |d| d.manifest.version + 1);
```

Comment:
This should be the latest `version + 1`?
For example, if you check out version 2 of `[1, 2, 3]`, and write again, it will overwrite the version 3, iiuc